lib/io/can.cpp

```diff
@@ -1,8 +1,74 @@
 #include "can.hpp"
 
+#include <stdio.h>
+#include <unistd.h>
+
+#include <linux/can.h>
+#include <net/if.h>
+#include <sys/ioctl.h>
+#include <sys/socket.h>
+
 namespace hyped::io {
-Can::Can()
+Can::Can(hyped::core::ILogger &logger) : logger_(logger)
+{
+}
+
+CanResult Can::initialiseCanSocket()
+{
+  if ((socket_ = socket(PF_CAN, SOCK_RAW, CAN_RAW)) < 0) {
+    logger_.log(hyped::core::LogLevel::kFatal, "Unable to open CAN socket");
+    return hyped::io::CanResult::kFailure;
+  }
+  sockaddr_can socket_address;
+  socket_address.can_family  = AF_CAN;
+  socket_address.can_ifindex = if_nametoindex("can1");
```

Comment:
Are we not working with any other can interfaces? If we are you should pass this to the function as an argument